frontend/src/components/CommentView.js

```diff
@@ -0,0 +1,275 @@
+import React, { useState } from 'react'
+import { Slate, Editable, withReact } from 'slate-react'
+import { createEditor } from 'slate'
+import { withHistory } from 'slate-history'
+import {
+    Text,
+    Image,
+    Box,
+    Popover,
+    PopoverTrigger,
+    PopoverContent,
+    HStack,
+} from '@chakra-ui/react'
+
+import { Element, Leaf } from './SlateComponents'
+import CommentInput from './CommentInput'
+
+const timeAgo = (dateString) => {
+    const date = new Date(dateString)
+    const now = new Date()
+    const seconds = Math.round((now - date) / 1000)
+    const minutes = Math.round(seconds / 60)
+    const hours = Math.round(minutes / 60)
+    const days = Math.round(hours / 24)
+    const weeks = Math.round(days / 7)
+    const months = Math.round(days / 30.44) // Average days per month
+    const years = Math.round(days / 365)
+
+    if (seconds < 60) {
+        return 'Just now'
+    } else if (minutes < 60) {
+        return `${minutes} minute${minutes > 1 ? 's' : ''} ago`
+    } else if (hours < 24) {
+        return `${hours} hour${hours > 1 ? 's' : ''} ago`
+    } else if (days < 7) {
+        return `${days} day${days > 1 ? 's' : ''} ago`
+    } else if (weeks < 4.35) {
```

Comment:
where does this 4.35 come from? is this supposed to represent 30 days in a month?